mesh/src/main/java/no/nordicsemi/android/mesh/transport/GenericBatteryStatus.java

```diff
@@ -40,9 +40,9 @@ void parseStatusParameters() {
         final ByteBuffer buffer = ByteBuffer.wrap(mParameters).order(ByteOrder.LITTLE_ENDIAN);
         mBatteryLevel = buffer.get();
         MeshLogger.verbose(TAG, "Battery level: " + mBatteryLevel);
-        if (buffer.limit() > GENERIC_BATTERY_STATUS_MANDATORY_LENGTH) {
-            mTimeToDischarge = buffer.get() | (buffer.get() << 8) | (buffer.get() << 16);
-            mTimeToCharge = buffer.get() | (buffer.get() << 8) | (buffer.get() << 16);
+        if (buffer.limit() >= GENERIC_BATTERY_STATUS_MANDATORY_LENGTH) {
```

Comment:
This check is not required as the status will be always 8 bytes,

Reply:
Should I remove this check though ? I thought it will be useful in case of a bad implementation in the embedded software.